desktop/ui/src/main/java/org/datacleaner/windows/FixedWidthDatastoreDialog.java

```diff
@@ -197,8 +189,60 @@ private void onSettingsUpdated(boolean autoDetectEncoding) {
 	}
 
 	@Override
+	protected byte[] getSampleBuffer() {
+		final File file = new File(getFilename());
+		final int bufferSize = getBufferSize();
+		byte[] bytes = new byte[bufferSize];
+		FileInputStream fileInputStream = null;
+
+		try {
+			fileInputStream = new FileInputStream(file);
+			int startPosition = getStartPosition();
+			fileInputStream.skip(startPosition);
+			int bytesRead = fileInputStream.read(bytes, 0, bufferSize);
+
+			if (bytesRead != -1 && bytesRead <= bufferSize) {
+				bytes = Arrays.copyOf(bytes, bytesRead);
+			}
+
+			return bytes;
+		} catch (IOException e) {
+			logger.error("IOException occurred while reading sample buffer", e);
+			return new byte[0];
+		} finally {
+			FileHelper.safeClose(fileInputStream);
+		}
+	}
+
+	private int getStartPosition() {
+		return _headerPresentCheckBox.isSelected() ? getRecordDataLength() : 0;
+	}
+
+	private int getBufferSize() {
+		return _eolPresentCheckBox.isSelected() ? SAMPLE_BUFFER_SIZE : getRecordDataLength();
+	}
+
+	private int getRecordDataLength() {
+		int length = 0;
+
+		if (_valueWidthTextFields != null && _valueWidthTextFields.size() > 0) {
+			for (JXTextField textField : _valueWidthTextFields) {
+				try {
+					final int columnWidth = Integer.parseInt(textField.getText());
+					length += columnWidth;
+				} catch (NumberFormatException e) {
+					// silently ignored
```

Comment:
Don't we want to pop up a "you screwed up"-kinda dialog? Or would it prevent entry at all?